File: frontend/lib/models/payment.dart
enum PaymentMethod { card, applePay, googlePay }
enum PaymentStatus { pending, paid, failed }

class Payment {
  final int id;
  final double amount;
  final PaymentMethod method;
  final PaymentStatus status;
  final DateTime createdAt;

  Payment({
    required this.id,
    required this.amount,
    required this.method,
    required this.status,
    required this.createdAt,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      id: json['id'],
      amount: json['amount'].toDouble(),
      method: _parseMethod(json['method']),
      status: _parseStatus(json['status']),
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  static PaymentMethod _parseMethod(String value) {
    switch (value) {
      case 'APPLE_PAY':
        return PaymentMethod.applePay;
      case 'GOOGLE_PAY':
        return PaymentMethod.googlePay;
      default:
        return PaymentMethod.card;
    }
  }

  static PaymentStatus _parseStatus(String value) {
    switch (value) {
      case 'PAID':
        return PaymentStatus.paid;
      case 'FAILED':
        return PaymentStatus.failed;
      default:
        return PaymentStatus.pending;
    }
  }
}

File: AirlineAIT/backend/app/models/payment.py
"""
Payment model with method and status tracking.

Stores payment transactions with method (CARD, APPLE_PAY, GOOGLE_PAY)
and status (PENDING, PAID, FAILED). Links to bookings.

Part of: Backend Models
"""
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey, Enum
from sqlalchemy.orm import relationship
import enum
from app.db.base import Base


class PaymentMethod(str, enum.Enum):
    CARD = "CARD"
    APPLE_PAY = "APPLE_PAY"
    GOOGLE_PAY = "GOOGLE_PAY"


class PaymentStatus(str, enum.Enum):
    PENDING = "PENDING"
    PAID = "PAID"
    FAILED = "FAILED"


class Payment(Base):
    __tablename__ = "payments"

    id = Column(Integer, primary_key=True, index=True)
    booking_id = Column(Integer, ForeignKey("bookings.id"), nullable=False)
    amount = Column(Float, nullable=False)
    method = Column(Enum(PaymentMethod), nullable=False)
    status = Column(Enum(PaymentStatus), default=PaymentStatus.PENDING, nullable=False)
    transaction_id = Column(String, unique=True, nullable=False, index=True)
    created_at = Column(DateTime, nullable=False)
    
    # Relationships
    booking = relationship("Booking", back_populates="payments")

